slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -133,25 +130,21 @@ def $name($args): $name = {
 
     trait PlainSqlMapperDef extends super.PlainSqlMapperDef{
       def code = {
-        def mkArgs(values: Seq[String] ) = {
-          if (values.size == 1) s"""( ${values.head} )"""
-          else values.mkString("(" ," , " , " )")
-        }
-
-        val positional = mkArgs(columnsPositional.map(c => (if(c.fakeNullable || c.model.nullable)s"<<?[${c.rawType}]"else s"<<[${c.rawType}]")))
+        def argList(values: Seq[String]) = values.mkString("(" ,", " , ")")
+        val positional = columnsPositional.map(c => (if(c.fakeNullable || c.model.nullable)s"<<?[${c.rawType}]"else s"<<[${c.rawType}]"))
         val dependencies = columns.map(_.exposedType).distinct.zipWithIndex.map{ case (t,i) => s"""e$i: GR[$t]"""}.mkString(", ")
-        val rearranged = mkArgs(desiredColumnOrder.map(i => tuple(i)) )
-        def result(args: String) = s"""${TableClass.elementType}${args}""" 
+        val rearranged = desiredColumnOrder.map(i => tuple(i))
+        def result(args: Seq[String]) = s"""${TableClass.elementType}${argList(args)}""" 
         val body =
-          if(autoIncLastAsOption && columns.size > 1 && columns.size <= 22 ){
+          if(autoIncLastAsOption && columns.size > 1){
             s"""
-val r = $positional
+val r = ${tupleBuilder(positional, "")}
 import r._
 ${result(rearranged)} // putting AutoInc last
```

Comment:
however this is still broken. It will generate `...(_1,...,_23,_25,...)`but should generate something like `...(_1._1,_1._2,...,_2._1,_2._2)` probably some room to unify the logic here with tupleBuilder or tupleFactory